counsel.el

```diff
@@ -647,10 +647,19 @@ When the selected variable is a `defcustom' with the type boolean
 or radio, offer completion of all possible values.
 
 Otherwise, offer a variant of `eval-expression', with the initial
-input corresponding to the chosen variable."
+input corresponding to the chosen variable.
+
+With a prefix arg, restrict list to variables defined using
+`defcustom'."
   (interactive (list (intern
                       (ivy-read "Variable: "
-                                (counsel-variable-list)
+                                (let ((vars (counsel-variable-list)))
+                                  (if current-prefix-arg
+                                      (cl-remove-if-not
+                                       (lambda (var)
+                                         (get (intern var) 'custom-type))
+                                       vars)
+                                    vars))
```

Comment:
AKA
```el
(ivy-read "Set variable: " (counsel-variable-list)
          :predicate (and current-prefix-arg
                          (lambda (varname)
                            (get (intern varname) 'custom-type)))
          :history 'counsel-set-variable-history
          :preselect (ivy-thing-at-point)
          :sort t
          :caller 'counsel-set-variable)
```

(Any reason `:sort` and `:caller` weren't already specified?)

Reply:
@basil-conto That version is nicer but it doesn't work for me.

Reply:
@justbur Can you please elaborate?

Reply:
@basil-conto Yes, sorry. With your version a prefix arg does not seem to have any affect on the list of candidates, while my initial version did eliminate candidates in my tests. I'm not sure why

Reply:
Oh, seems to be #1158.